pkg/k8s/apis/cilium.io/v2alpha1/cidrgroups_types.go

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package v2alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/cilium/cilium/pkg/policy/api"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:categories={cilium},singular="ciliumcidrgroup",path="ciliumcidrgroups",scope="Cluster",shortName={ccg}
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +deepequal-gen=false
+
+// CiliumCIDRGroup is a list of CIDRs that can be referenced as a single entity
+// from CiliumNetworkPolicies.
+type CiliumCIDRGroup struct {
+	// +deepequal-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +deepequal-gen=false
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +kubebuilder:validation:Required
+	Spec CiliumCIDRGroupSpec `json:"spec"`
+}
+
+type CiliumCIDRGroupSpec struct {
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=0
+	CIDRs []api.CIDR `json:"cidrs"`
```

Comment:
Given that these CIDRs end up only selecting peers outside the cluster, I'd suggest naming something like this:

```suggestion
	ExternalCIDRs []api.CIDR `json:"external-cidrs"`
```

This would make the semantics a bit more clear when they interact with other resources like CNP, since users are less likely to assume that CIDRGroupRefs also select CIDRs for peers inside the cluster.

Reply:
(We could probably elaborate a bit more as well in comments, to make these semantics more obvious for users)